label_studio/ml/serializers.py

```diff
@@ -2,50 +2,119 @@
+        """ """
+        healthcheck_response = MLBackend.healthcheck_(**attrs)
+
         if healthcheck_response.is_error:
-            raise serializers.ValidationError(
-                f"Can't connect to ML backend {url}, health check failed. "
-                f'Make sure it is up and your firewall is properly configured. '
-                f'<a href="https://labelstud.io/guide/ml.html">Learn more</a>'
-                f' about how to set up an ML backend. Additional info:' + healthcheck_response.error_message
-            )
-        project = attrs['project']
-        setup_response = MLBackend.setup_(url, project)
+            if healthcheck_response.status_code == 401:
+                message = (
+                    'Able to connect to ML Server, but authentication parameters were '
+                    'either not provided or are incorrect.'
+                )
+            else:
+                message = (
+                    f"Can't connect to ML backend {attrs['url']}, health check failed. "
+                    'Make sure it is up and your firewall is properly configured. '
+                    f'<a href="https://labelstud.io/guide/ml.html">Learn more</a> '
+                    f'about how to set up an ML backend. Additional info: {healthcheck_response.error_message}'
+                )
+
+            raise serializers.ValidationError(message)
+
+    def _validate_setup(self, attrs):
+        """ """
+        setup_response = MLBackend.setup_(**attrs)
+
         if setup_response.is_error:
-            raise serializers.ValidationError(
-                f"Successfully connected to {url} but it doesn't look like a valid ML backend. "
+            message = (
+                f"Successfully connected to {attrs['url']} but it doesn't look like a valid ML backend. "
                 f'Reason: {setup_response.error_message}.\n'
-                f'Check the ML backend server console logs to check the status. There might be\n'
-                f'something wrong with your model or it might be incompatible with the current labeling configuration.'
+                'Check the ML backend server console logs to check the status.'
+                'There might be something wrong with your model or it might be incompatible with the current labeling configuration.'
             )
+
+            raise serializers.ValidationError(message)
+
+    def validate(self, attrs):
+        """ """
+        attrs = super().validate(attrs)
+
+        self._validate_authentication(attrs)
+        self._validate_healthcheck(attrs)
+        self._validate_setup(attrs)
+
         return attrs
 
     class Meta:
         model = MLBackend
-        fields = '__all__'
+        fields = [
+            'id',
+            'state',
+            'readable_state',
+            'is_interactive',
+            'url',
+            'error_message',
+            'title',
+            'auth_method',
+            'basic_auth_user',
+            'basic_auth_pass',
```

Comment:
We need to make sure that these fields are writable but not readable in the API.

Reply:
ok, I do see this is write_only: https://github.com/HumanSignal/label-studio/pull/5570/files#diff-a67b82b0de34ce236f0f9a212e530d2a865c73f603e0e5e6ca485d5c0ea45d10R13

would be great to capture this in a test though.